flow/Trace.cpp

```diff
@@ -783,6 +783,13 @@ TraceEvent::TraceEvent(TraceEvent &&ev) {
 	tmpEventMetric = ev.tmpEventMetric;
 	trackingKey = ev.trackingKey;
 	type = ev.type;
+	timeIndex = ev.timeIndex;
+
+	for (int i = 0; i < 5; i++) {
```

Comment:
Well, I guess this magic number is really confusing me... I am fixing it in https://github.com/apple/foundationdb/pull/4003/files

Reply:
Thank you.  We only use this (and operator==(&&) in one file.  It might make sense to delete them (maintaining them is error prone, and they're only useful when you want to start building a  trace event, then wait(), then continue to build the trace event.

Even there, it seems like a bug (or at least surprising behavior) in the flow compiler.  `state Foo foo(...);` should not invoke `Foo(&&)` or `operator=(Foo&&)`.